lib/app_profiler/base_profile.rb

```diff
@@ -34,7 +35,8 @@ def from_vernier(data)
     # `data` is assumed to be a Hash for Stackprof,
     # a vernier "result" object for vernier
     def initialize(data, id: nil, context: nil)
-      @id      = id.presence || ProfileId.current
+      ProfileId.set(id) if id.present?
```

Comment:
We were not doing this before. If someone provides an Id we should set ProfileId too so that the values are consistent.